src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -39071,6 +39071,200 @@ static bool G(out object o)
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "(c, null)").WithArguments("(C<string?> c, string?)", "(C<string>, string)").WithLocation(14, 51),
+                // (15,9): warning CS8602: Possible dereference of a null reference.
+                //         t2.Item1.ToString(); // warn
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t2.Item1").WithLocation(15, 9),
+
+                // line 3
+                // (19,44): warning CS8619: Nullability of reference types in value of type '(string?, (C<string>, string))' doesn't match target type '(string, (C<string>, string))'.
+                //         (string, (C<string>, string)) t3 = (null, (c, null)!);
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "(null, (c, null)!)").WithArguments("(string?, (C<string>, string))", "(string, (C<string>, string))").WithLocation(19, 44),
+                // (20,9): warning CS8602: Possible dereference of a null reference.
+                //         t3.Item1.ToString(); // warn
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t3.Item1").WithLocation(20, 9),
+
+                // line 4
+                // (25,9): warning CS8602: Possible dereference of a null reference.
+                //         t4.Item1.ToString(); // warn
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t4.Item1").WithLocation(25, 9),
+
+                // line 6
+                // (34,52): warning CS8619: Nullability of reference types in value of type '(C<string?>, string)' doesn't match target type '(C<string>, string)'.
+                //         (string, (C<string>, string)) t6 = (null!, (c!, null!)); // warn
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "(c!, null!)").WithArguments("(C<string?>, string)", "(C<string>, string)").WithLocation(34, 52),
+
+                // line 8
+                // (44,51): warning CS8619: Nullability of reference types in value of type '(C<string?> c, string?)' doesn't match target type '(C<string>, string)'.
+                //         (string, (C<string>, string)) t8 = (null, (c, null))!; // warn
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "(c, null)").WithArguments("(C<string?> c, string?)", "(C<string>, string)").WithLocation(44, 51),
+                // (45,9): warning CS8602: Possible dereference of a null reference.
+                //         t8.Item1.ToString(); // warn
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t8.Item1").WithLocation(45, 9));
+
+            CompileAndVerify(comp);
+        }
+
+        [Fact, WorkItem(31370, "https://github.com/dotnet/roslyn/issues/31370")]
+        public void SuppressNullableWarning_TupleEquality()
+        {
+            var source =
+@"class C<T>
+{
+    static void M((string, C<string>) t, C<string?> c)
+    {
+        _ = t == (null, c);
```

Comment:
>== [](start = 14, length = 2)

Could you add permutations where the suppression is on the LHS?
`(t, (null, c)!) == ((null, c)!, t` for example